opendata_module/opmon_opendata/api/helpers.py

```diff
@@ -0,0 +1,213 @@
+import json
+from datetime import datetime
+from gzip import GzipFile
+from io import BytesIO
+from typing import Generator, List, Optional, Tuple, TypedDict
+
+import psycopg2
+from dateutil import relativedelta
+from django.core.handlers.wsgi import WSGIRequest
+
+from opmon_opendata.api.input_validator import OpenDataInputValidator
+from opmon_opendata.api.postgresql_manager import PostgreSQL_Manager
+
+DEFAULT_STREAM_BUFFER_LINES = 1000
+
+
+class OrderByType(TypedDict):
+    column: str
+    order: str
+
+
+def generate_ndjson_stream(postgres: PostgreSQL_Manager, date: datetime,
+                           columns: List[str], constraints: List[dict],
+                           order_clauses: List[dict], settings: dict) -> Generator[bytes, None, None]:
+    """Creates a gzipped ndjson file iterator suitable for StreamingHttpResponse."""
```

Comment:
```suggestion
    """Creates a gzipped json file iterator suitable for StreamingHttpResponse."""
```

Reply:
http://ndjson.org format is used here. Technically is it json, but could be worth noting the format.